spec-tests/src/test/java/org/candlepin/spec/content/ContentAccessSpecTest.java

```diff
@@ -805,6 +812,85 @@ public void shouldReturnANotModifiedReturnCodeWhenTheDataHasNotBeenUpdatedSinceD
             .getContentAccessBodyJson(consumer.getUuid(), formatted));
     }
 
+    @Test
+    public void shouldReturnServiceUnavailableWithConcurrentPayloadCreation() throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.randomSca());
+        String ownerKey = owner.getKey();
+
+        ProductDTO prod = adminClient.ownerProducts().createProduct(ownerKey, Products.random());
+        ContentDTO content = adminClient.ownerContent().createContent(ownerKey, Contents.random());
+        adminClient.ownerProducts()
+            .addContentToProduct(ownerKey, prod.getId(), content.getId(), true);
+        adminClient.owners().createPool(ownerKey, Pools.random(prod));
+
+        ConsumerDTO consumer1 = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer2 = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer3 = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer4 = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer5 = adminClient.consumers().createConsumer(Consumers.random(owner));
+
+        List<ConsumerDTO> consumers = List.of(consumer1, consumer2, consumer3, consumer4, consumer5);
+
+        int maxNumberOfAttempts = 5;
```

Comment:
This test usually creates the `PersistenceException` after 1 or 2 attempts for me locally, so I bumped the number of attempts up to 5 so that the test should consistently reproduce the exception.